utils/SnapPy/Snappy/EEMEP/PostProcess.py

```diff
@@ -56,46 +107,88 @@ def convert_files(self, instantFilename, averageFilename):
         os.rename(averageFilename, newAvgFile)
 
 
-class Accumulate:
-    def __init__(self, path, logger) -> None:
-        self.path = path
 
-        if logger:
-            self.logger = logger
-        else:
-            self.logger = StderrLogger()
 
-    def accumulate_file(self, filename) -> None:
-        self.logger._write_log(f"Accumulating outputs found in {self.path}")
+    def accumulate_and_toa_nuc_files(self, instantFilename, averageFilename) -> None:
+        self.logger._write_log(f"Accumulating nuclear outputs found in {self.path}")
+        # rename files, make them available to further processes
+        timestamp = self.timestamp.strftime("%Y%m%dT%H%M%S")
+        simulationstart = self.timestamp.strftime("%Y-%m-%d_%H:%M:%S")
 
         dir = Path(self.path)
 
-        new_filename = filename.split(".")[0] + "_acc.nc"
-
-        nc_path = dir / filename
-        new_path = dir / new_filename
-
-        shutil.copy(nc_path, new_path)
-
-        with Dataset(new_path, 'a') as nc:
-            for var in list(nc.variables):
-                if var not in ['lon', 'lat', 'lev', 'P0', 'hyam', 'hybm', 'ilev', 'hyai', 'hybi', 'time', 'PS']:
-
-                    self.logger._write_log(f"Accumulating {var}")
-                    acc = nc.createVariable(f'{var}_ACC','f4',('time','lat','lon'), zlib=True)
-
-                    acc_data = np.cumsum(nc[var][:], axis=0)
-                    unit = nc[var].units
-                    unit_new = unit.split("/")[0] + " h/" + unit.split("/")[1]
-
-                    acc.units = unit_new
-                    acc.coordinates = nc[var].coordinates
-
-                    acc[:] = acc_data
-
-            nc.sync()
+        new_filename = dir / 'eemep_nuc_{}.nc'.format(timestamp)
+        nc_path = dir / instantFilename
+        shutil.copy(nc_path, new_filename)
 
+        isotopes = set()
+        with netCDF4.Dataset(new_filename, 'a') as nc:
+            nc.setncattr('SIMULATION_START_DATE', simulationstart)
+            for varname, var in nc.variables.items():
+                if varname.startswith('SURF_uBq_NPP_'):
+                    isotopes.add(var.name[13:])
+            # rename variables in copied instantFile
+            for isotope in isotopes:
+                varname = f'SURF_uBq_NPP_{isotope}'
+                newvar = f'{isotope}_concentration'
+                if varname in nc.variables:
+                    self.logger._write_log(f"renaming {varname} to {newvar}")
+                    # nc.renameVariabe(varname, newvar) # bug in renameVariable -> just create a copy for now
+                    var = nc[varname]
+                    nvar = nc.createVariable(newvar, var.datatype, dimensions=var.dimensions,
+                                            zlib=True, complevel=1)
+                    nvar.setncatts(var.__dict__)
+                    for t in range(0, nc['time'][:].shape[0]):
```

Comment:
`range(nc["time"].shape[0])`

Reply:
better readable here, in particular since I use in other places `for t in range(1, nc['time'][:].shape[0]):`

And this code only works around a bug in netCDF4.Dataset.renameDataset.

Reply:
At least read only the shape and not the entire dataset: `[:]` should be removed

Reply:
ok